### PR TITLE
[DJC-001] Setup Devise JWT controller gem

### PR DESCRIPTION
**Pull request by holywyvern**
## Summary

This just sets up the environment to do the gem, installs dependencies and such.
Shouldn't be too complex.

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+2.6.3
```
Comment (blacksam07) [inline .ruby-version]: why don't use 2.6.5 ?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+sudo: false
+language: ruby
+cache: bundler
+rvm:
+  - 2.6.2
```
Comment (blacksam07) [inline .travis.yml]: in ruby-version file they are using 2.6.3

File: devise_jwt_controllers.gemspec [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+require 'devise/jwt/controllers/version'
+
+Gem::Specification.new do |spec| # rubocop:disable Metrics/BlockLength
+  spec.name          = 'devise_jwt_controllers'
+  spec.version       = Devise::JWT::Controllers::VERSION
+  spec.authors       = ['Ramiro Rojo']
+  spec.email         = ['ramiro.rojo@wolox.com.ar']
+
+  spec.summary       = 'A gem to generate documentation out of tests.'
+  spec.homepage      = 'https://github.com/holywyvern/fictium'
```
Comment (blacksam07) [inline devise_jwt_controllers.gemspec]:
```suggestion
  spec.homepage      = 'https://github.com/Wolox/devise_jwt_controllers'
```

File: devise_jwt_controllers.gemspec [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  spec.license       = 'Apache-2.0'
+
+  spec.metadata['homepage_uri'] = spec.homepage
+  spec.metadata['source_code_uri'] = 'https://github.com/holywyvern/fictium'
```
Comment (blacksam07) [inline devise_jwt_controllers.gemspec]:
```suggestion
  spec.metadata['source_code_uri'] = 'https://github.com/Wolox/devise_jwt_controllers'
```

File: devise_jwt_controllers.gemspec [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  spec.metadata['changelog_uri'] = 'https://github.com/holywyvern/fictium/CHANGELOG.md'
```
Comment (blacksam07) [inline devise_jwt_controllers.gemspec]:
```suggestion
  spec.metadata['changelog_uri'] = 'https://github.com/Wolox/devise_jwt_controllers/CHANGELOG.md'
```

File: .ruby-version (hunk 1 above)
Comment (holywyvern) [reply .ruby-version]: Ruby 2.6.5 was released *today* I don't know if travis-ci has support for it yet.

File: devise_jwt_controllers.gemspec [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+require 'devise/jwt/controllers/version'
+
+Gem::Specification.new do |spec| # rubocop:disable Metrics/BlockLength
+  spec.name          = 'devise_jwt_controllers'
+  spec.version       = Devise::JWT::Controllers::VERSION
+  spec.authors       = ['Ramiro Rojo']
+  spec.email         = ['ramiro.rojo@wolox.com.ar']
+
+  spec.summary       = 'A gem to generate documentation out of tests.'
+  spec.homepage      = 'https://github.com/Wolox/fictium'
```
Comment (blacksam07) [inline devise_jwt_controllers.gemspec]:
```suggestion
  spec.homepage      = 'https://github.com/Wolox/devise_jwt_controllers'
```

File: devise_jwt_controllers.gemspec [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  spec.license       = 'Apache-2.0'
+
+  spec.metadata['homepage_uri'] = spec.homepage
+  spec.metadata['source_code_uri'] = 'https://github.com/Wolox/fictium'
+  spec.metadata['changelog_uri'] = 'https://github.com/Wolox/fictium/CHANGELOG.md'
```
Comment (blacksam07) [inline devise_jwt_controllers.gemspec]: same error

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+---
+sudo: false
+language: ruby
+cache: bundler
+rvm:
```
[the remark this answers is not in the document]
Comment (holywyvern) [reply .travis.yml]: The gem won't give support to any ruby version older than 2.6:

```
spec.required_ruby_version = '>= 2.6.0'
```

File: README.md [hunk 9]
```diff
@@ -38,6 +40,17 @@ Bug reports and pull requests are welcome on GitHub at https://github.com/[USERN
 
 The gem is available as open source under the terms of the [MIT License](https://opensource.org/licenses/MIT).
 
+## About
+This project was developed by [Leon Calermo](leoncalermo) and [Ramiro Rojo](https://github.com/leoncalermo).
```
Comment (blacksam07) [inline README.md]:
```suggestion
This project was developed by [Leon Calermo](leoncalermo) and [Ramiro Rojo](https://github.com/holywyvern).
```

File: .gitignore [hunk 10]
```diff
@@ -0,0 +1,15 @@
+/.bundle/
+/.yardoc
+/_yardoc/
+/coverage/
+/doc/
+/pkg/
+/spec/reports/
+/tmp/
+/log/
+.DS_Store
```
Comment (lucasVoboril) [inline .gitignore]: :confounded: this file is in GitHub 
image

File: config/database.yml [hunk 11]
```diff
@@ -0,0 +1,25 @@
+# SQLite version 3.x
+#   gem install sqlite3
+#
+#   Ensure the SQLite 3 gem is defined in your Gemfile
+#   gem 'sqlite3'
+#
+default: &default
+  adapter: sqlite3
```
Comment (lucasVoboril) [inline config/database.yml]: This adapter is ok?
```suggestion
  adapter: postgresql
```

Comment (lucasVoboril) [reply config/database.yml]: enconding: unicode

File: lib/devise/jwt/extensions/warden_helper.rb [hunk 12]
```diff
@@ -0,0 +1,30 @@
+module Devise
+  module JWT
+    module WardenHelper
```
Comment (lucasVoboril) [inline lib/devise/jwt/extensions/warden_helper.rb]: :smile: :clap: nice

File: config/database.yml (hunk 11 above)
Comment (holywyvern) [reply config/database.yml]: Sqlite is ok for a gem test, let's avoid to add extra development dependencies if they are not needed.

File: spec/rails/config/initializers/devise.rb [hunk 13]
```diff
@@ -0,0 +1,28 @@
+Devise.setup do |config|
+  config.secret_key = '57b5d0a02bc1e025c5077d9a62d329f66432ba975d3'
```
Comment (nicozar95) [inline spec/rails/config/initializers/devise.rb]: Until I read this was in spec 😆

**Pull request merged**